code/modules/mob/living/update_status.dm

```diff
@@ -75,45 +64,86 @@
 	return eye_blurry
 
 //Updates canmove, lying and icons. Could perhaps do with a rename but I can't think of anything to describe it.
-/mob/living/update_canmove(delay_action_updates = 0)
-	var/fall_over = !can_stand()
-	var/buckle_lying = !(buckled && !buckled.buckle_lying)
-	if(fall_over || resting || stunned)
-		drop_r_hand()
-		drop_l_hand()
+/mob/living/proc/update_mobility()
```

Comment:
here is basically how the new mobility stuff is handled, replaced update_canmove with update_mobility.